scroll-animations-1/Overview.bs

```diff
@@ -602,3 +653,146 @@ spec: cssom-view-1; type: dfn;
+	then any keyframes attached to points on that [=named timeline range=]
+	are ignored.
+
+## Attaching Animations to Timeline Ranges ## {#named-range-animation-declaration}
+
+	A set of animation keyframes can be attached
+	in reference to [=named timeline ranges=]
+	by adjusting the duration of the animation.
+
+	<pre class="propdef shorthand">
+		Name: animation-delay, animation-range
+		Value: [ <<'animation-delay-start'>> <<'animation-delay-end'>>? | <<timeline-range-name>> ]#
+	</pre>
+
+	The 'animation-delay' and 'animation-range' properties
+	are both [=shorthand property|shorthands=]
+	that set 'animation-delay-start' and 'animation-delay-end'
+	together in a single declaration.
+	If the <<'animation-delay-end'>> value is omitted,
+	it is set to zero.
+	If a <<timeline-range-name>> alone is specified,
+	'animation-delay-start' is set to that name plus ''0%''
+	and 'animation-delay-end' is set to that name plus ''100%''.
+
+	<pre class="propdef">
+		Name: animation-delay-start
+		Value: [ <<time>> | <<timeline-range-name>> <<percentage>>
+		Initial: 0s
+		Applies to: all elements
+		Inherited: no
+		Percentages: relative to the specified [=named timeline range=]
+		Computed value: list, each item either a duration or a timeline range and progress percentage
+		Animatable: no
+	</pre>
+
+	Defines a delay between the start of the animation
+	and when it begins executing
+	(i.e. when keyframes attached to 0% progress are mapped).
+
+	<dl dfn-for="animation-delay-start, animation-delay-end" dfn-type=value>
+		<dt><dfn><<time>></dfn>
+		<dd>
+			Defines the delay as a duration. See [[CSS-ANIMATIONS-1]].
+		<dt><dfn><<timeline-range-name>> <<percentage>></dfn>
+		<dd>
+			Calculates the delay as the difference between
+			the start of the animation
+			and the specified point on the [=timeline=].
+	</dl>
+
+	<pre class="propdef">
+		Name: animation-delay-end
+		Value: [ <<time>> | <<timeline-range-name>> <<percentage>>
+		Initial: 0s
+		Applies to: all elements
+		Inherited: no
+		Percentages: relative to the specified [=named timeline range=]
+		Computed value: list, each item either a duration or a timeline range and progress percentage
+		Animatable: no
+	</pre>
+
+	Defines a delay between the end of the animation
+	and when it ends executing
+	(i.e. when keyframes attached to 100% progress are mapped).
+	Values are as for 'animation-delay-start',
+	but calculating backwards from the end,
+	i.e. a ''2s'' delay effectively shortens the animation duration,
+	whereas ''-2s'' lengthens it.
+
+## Reporting Timeline Range Progress: the ''getTime()'' method ## {#named-range-getTime}
+
+	Progress through named ranges is exposed on the {{AnimationTimeline}} object
+	by the {{AnimationTimeline/getCurrentTime()}} method:
+
+	<pre class="webidl">
+		[Exposed=Window]
+		interface AnimationTimeline {
+		    readonly attribute double? currentTime;
+		    CSSNumericValue getCurrentTime(CSSOMString? rangeName);
```

Comment:
You want `optional CSSOMString`, not `CSSOMString?` - the latter means the argument is still required, but you can pass `null`.

Reply:
Ah, and it looks like (from your algo) that you want it to *return* a `CSSNumericValue?`.